schema/schema.go

```diff
@@ -35,23 +40,27 @@ func (s *WebRPCSchema) Validate() error {
 		return fmt.Errorf("webrpc schema version, '%s' is invalid, try '%s'", s.WebrpcVersion, SCHEMA_VERSION)
 	}
 
-	for _, msg := range s.Types {
-		err := msg.Parse(s)
+	for i := 0; i < len(s.Types); i++ {
+		err := s.Types[i].Parse(s, i)
 		if err != nil {
-			return err
+			if !errors.Is(err, errAlreadyParsed) {
```

Comment:
Do we have to pass the index into the Parse() function? Feels like a hack to me.

What does `errAlreadyParsed` mean? Is it that
1) the type was already parsed and was identical?
2) the type of a same name already exists (but is not necessarily the same)?

Reply:
1. same name + file/line